### DISCUSSION TITLE
Can egui be used to make transparent overlay addon program over a game?

### DISCUSSION THREAD

**Discussion by coderedart** [emilk/egui#354] [Q&A] [answered]
Apologies for the long question. 
I am thinking of making a crossplatform overlay program for a game called guild wars 2. something similar to http://www.gw2taco.com/2015/01/feature-list.html which is windows only. 

in terms of GUI, there's 2 requirements for doing this.

1. The main window needs to be transparent and stay on top with only the markers or whatever i draw on the painter device being visible over the game. 
2. The main window must allow all the inputs to pass through like mouse cursor clicks/keyboard key presses etc.. *except* a file menu on top left which must open the menu/settings for the program when clicked

i guess in terms of Qt, i want to be able to set the widget attributes Qt::WA_TranslucentBackground for transparency and the window attributes Qt::WindowStaysOnTopHint to keep it ontop of game and Qt::WindowTransparentForInput to allow input to pass through and finally Qt::FramelessWindowHint so that there's no minimize/close buttons or titlebar as the overlay should not be able to move;  

I just wanna know if this is possible in egui, as I really want want to use it over qt. the gui getting input from the socket/game shared memory, and drawing the routes/markers etc.. every frame seems like a job for immediate mode api.

**Comment by emilk** [comment, marked as answer]
`egui_glium` already supports [transparent](https://docs.rs/epi/0.11.0/epi/trait.App.html#method.transparent) and [non-decorated](https://docs.rs/epi/0.11.0/epi/trait.App.html#method.decorated) windows. Adding a `always_on_top` flag is easy.
Passing on input should probably be doable, but I'm not sure how (help wanted?).

**Reply by emilk** [reply]
I added `always_on_top` as an option in a7a36bd3137ecbcf7a8b452596a18481c6707050
